### PR TITLE
ref(js): Do not use `-c` with eslint

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Overriding the config file breaks the cascading features of eslint, meaning the .eslintrc.js file we have inside `docs-ui` won't be read [0] when running linting in CI. See [1]

[0]: This is why https://github.com/getsentry/sentry/pull/32899 is failing even after https://github.com/getsentry/sentry/pull/32897

[1]: https://eslint.org/docs/user-guide/configuring/configuration-files#cascading-and-hierarchy

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
+/* eslint-env node */
+/* eslint import/no-nodejs-modules:0 */
+
+const process = require('process');
+
+const isRelxed = !!process.env.SENTRY_ESLINT_RELAXED;
```
Comment (JonasBa) [inline .eslintrc.js:6]:
```suggestion
const isRelaxed = !!process.env.SENTRY_ESLINT_RELAXED;
```

File: .eslintrc.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+
 module.exports = {
-  extends: ['sentry-app/strict'],
+  extends: [isRelxed ? 'sentry-app' : 'sentry-app/strict'],
```
Comment (JonasBa) [inline .eslintrc.js:9]:
```suggestion
  extends: [isRelaxed ? 'sentry-app' : 'sentry-app/strict'],
```

**Review by JonasBa — APPROVED** [review]
Small typo, but code looks good! Btw, why do we not just lint everything with strict config here? Do we use relaxed for perf? I remember finding that slightly confusing when I joined, maybe we could document that somewhere

**Comment by evanpurkhiser** [comment]
We use the `strict` linting for lints that might be in progress. So say we wanted to start enforcing no default exports (this might be a bad example bc it's work), we would add that to our strict ruleset, requiring people to update it for it to pass strict, but in CI we wouldn't lint for it since it wouldn't be fixed everywhere.

**Review by billyvg — APPROVED** [review]
This looks good, `js-build-and-lint` should be the only workflow that needs strict checking. If we need to add "relaxed" rules, we'd add them to `sentry-app` as warnings.

**Pull request merged**